unhandled panic from workflow code

Panic from workflow code, including any code running in corotine created by any workflow.Go() should be handled by sdk.
User reported this panic stack trace suggest that we might have panic coming from [here](https://github.com/temporalio/sdk-go/blob/384e906eaace5e8914a6c0890419c23eba3b033c/internal/internal_workflow.go#L863)
panic: manual inject panic from go routine [recovered]
	panic: runtime error: slice bounds out of range [9:8]

goroutine 52489 [running]:
go.temporal.io/sdk/internal.getStackTraceRaw({0xc000e7c180, 0x14}, 0x9, 0x4)
	external/io_temporal_go_sdk/internal/internal_workflow.go:863 +0x1bd
go.temporal.io/sdk/internal.getStackTrace({0x52ba998, 0x7f069a39aa68}, {0x30317d5, 0x5}, 0x4)
	external/io_temporal_go_sdk/internal/internal_workflow.go:853 +0xab
go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine.func1.1()
	external/io_temporal_go_sdk/internal/internal_workflow.go:936 +0x73
panic({0x2995340, 0x346e360})
	GOROOT/src/runtime/panic.go:1038 +0x215
xxxxxxx/xxxx/xxxxxxxx.(*Workflow).startBlocks.func1({0xc001857950, 0x1})
	xxxxx/xxxxxxxx/workflow.go:432 +0x27
go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine.func1(0xc0008ee120)
	external/io_temporal_go_sdk/internal/internal_workflow.go:941 +0xde
created by go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine
	external/io_temporal_go_sdk/internal/internal_workflow.go:932 +0x365
panic: manual inject panic from go routine [recovered]
	panic: runtime error: slice bounds out of range [9:8]

goroutine 52382 [running]:
go.temporal.io/sdk/internal.getStackTraceRaw({0xc001e65698, 0x14}, 0x9, 0x4)
	external/io_temporal_go_sdk/internal/internal_workflow.go:863 +0x1bd
go.temporal.io/sdk/internal.getStackTrace({0x52ba998, 0x34ed810}, {0x30317d5, 0x5}, 0x4)
	external/io_temporal_go_sdk/internal/internal_workflow.go:853 +0xab
go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine.func1.1()
	external/io_temporal_go_sdk/internal/internal_workflow.go:936 +0x73
panic({0x2995340, 0x346e360})
	GOROOT/src/runtime/panic.go:1038 +0x215
xxxxxxx/xxxxxxx/xxxxxxxxxx.(*Workflow).startBlocks.func1({0xc001c8fe50, 0x1})
	xxxxxxx/xxxxxxxxxx/workflow.go:432 +0x27
go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine.func1(0xc0024b1030)
	external/io_temporal_go_sdk/internal/internal_workflow.go:941 +0xde
created by go.temporal.io/sdk/internal.(*dispatcherImpl).NewCoroutine
	external/io_temporal_go_sdk/internal/internal_workflow.go:932 +0x365

--- a/internal/internal_workflow.go
+++ b/internal/internal_workflow.go
@@ -860,10 +860,11 @@ func getStackTraceRaw(top string, omitTop, omitBottom int) string {
 		return rawStack
 	}
 	lines := strings.Split(rawStack, "\n")
-	// If the top is after the end, set as same
 	omitEnd := len(lines) - omitBottom
+	// If the start is after the end, the depth was invalid originally so return
+	// the entire raw stack
 	if omitTop > omitEnd {
-		omitTop = omitEnd
+		return rawStack
 	}
 	lines = lines[omitTop:omitEnd]
 	lines = append([]string{top}, lines...)

--- a/internal/internal_workflow.go
+++ b/internal/internal_workflow.go
@@ -860,7 +860,12 @@ func getStackTraceRaw(top string, omitTop, omitBottom int) string {
 		return rawStack
 	}
 	lines := strings.Split(rawStack, "\n")
-	lines = lines[omitTop : len(lines)-omitBottom]
+	// If the top is after the end, set as same
+	omitEnd := len(lines) - omitBottom
+	if omitTop > omitEnd {
+		omitTop = omitEnd
+	}
+	lines = lines[omitTop:omitEnd]
 	lines = append([]string{top}, lines...)
 	return strings.Join(lines, "\n")
 }

--- a/internal/internal_workflow_test.go
+++ b/internal/internal_workflow_test.go
@@ -1440,8 +1440,8 @@ func TestStackTraceInvalidDepthBounded(t *testing.T) {
 	// Confirm at 2 depth there are 3 lines (1 for header, 2 for fn and path)
 	lines := strings.Split(getStackTrace("mycoroutine", "success", 2), "\n")
 	require.Len(t, lines, 3)
-	// But at invalid 100 depth, there is only 1 header line because bound is
-	// trimmed
+	// But at invalid 100 depth, there are more than three lines but less than 100
+	// because we show the full trace when bounds are wrong
 	lines = strings.Split(getStackTrace("mycoroutine", "success", 100), "\n")
-	require.Len(t, lines, 1)
+	require.True(t, len(lines) > 3 && len(lines) < 100)
 }

--- a/internal/internal_workflow_test.go
+++ b/internal/internal_workflow_test.go
@@ -1435,3 +1435,13 @@ func (t *tracingOutboundCallsInterceptor) ExecuteActivity(ctx Context, activityT
 	t.inbound.trace = append(t.inbound.trace, "ExecuteActivity "+activityType)
 	return t.Next.ExecuteActivity(ctx, activityType, args...)
 }
+
+func TestStackTraceInvalidDepthBounded(t *testing.T) {
+	// Confirm at 2 depth there are 3 lines (1 for header, 2 for fn and path)
+	lines := strings.Split(getStackTrace("mycoroutine", "success", 2), "\n")
+	require.Len(t, lines, 3)
+	// But at invalid 100 depth, there is only 1 header line because bound is
+	// trimmed
+	lines = strings.Split(getStackTrace("mycoroutine", "success", 100), "\n")
+	require.Len(t, lines, 1)
+}

